util/azure/utils.go

```diff
@@ -0,0 +1,352 @@
+	operationsMap := NewOperationsMap()
+
+	for _, resList := range apiResourcesList {
+		if len(resList.APIResources) == 0 {
+			continue
+		}
+
+		group := "v1" // core api group
+		if resList.GroupVersion != "" && resList.GroupVersion != "v1" {
+			group = strings.Split(resList.GroupVersion, "/")[0]
+		}
+
+		for _, apiResource := range resList.APIResources {
+			if strings.Contains(apiResource.Name, "/") {
+				continue
+			}
+
+			actionId := clusterType
+			if group != "v1" {
+				actionId = path.Join(actionId, group)
+			}
+
+			resourceName := apiResource.Name
+
+			actionId = path.Join(actionId, resourceName)
+
+			for _, operation := range operationsList {
+				if strings.Contains(operation.Name, actionId) {
+					opNameArr := strings.Split(operation.Name, "/")
+
+					if group != "v1" {
+						// extra validation to make sure groups are the same
+						if group != opNameArr[2] {
+							continue
+						}
+					} else {
+						// make sure resources are the same for core apigroup
+						if resourceName != opNameArr[2] {
+							continue
+						}
+					}
+
+					verb := opNameArr[len(opNameArr)-1]
+					if verb == "action" {
+						verb = path.Join(opNameArr[len(opNameArr)-2], opNameArr[len(opNameArr)-1])
+					}
+
+					da := DataAction{
+						ActionInfo: AuthorizationActionInfo{
+							IsDataAction: true,
+						},
+						IsNamespacedResource: apiResource.Namespaced,
+					}
+					da.ActionInfo.AuthorizationEntity.Id = operation.Name
+
+					if _, found := operationsMap[group]; !found {
+						operationsMap[group] = NewResourceAndVerbMap()
+					}
+
+					if _, found := operationsMap[group][resourceName]; !found {
+						operationsMap[group][resourceName] = NewVerbAndActionsMap()
+					}
+
+					operationsMap[group][resourceName][verb] = da
+				}
+			}
+		}
+	}
+
+	return operationsMap
+}
+
+func fetchApiResources(kubeconfigFilePath string) ([]*metav1.APIResourceList, error) {
+	// creates the in-cluster config
+	klog.V(5).Infof("Fetching list of APIResources from the apiserver.")
+
+	var cfg *rest.Config
+	var err error
+	if kubeconfigFilePath != "" {
+		cfg, err = clientcmd.BuildConfigFromFlags("", kubeconfigFilePath)
+	} else {
+		cfg, err = rest.InClusterConfig()
+	}
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubeconfig")
+	}
+
+	kubeclientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error building kubernetes clientset")
+	}
+
+	apiresourcesList, err := kubeclientset.Discovery().ServerPreferredResources()
+	if err != nil {
+		return nil, err
+	}
+
+	if klog.V(5).Enabled() {
+		printApiresourcesList, _ := json.Marshal(apiresourcesList)
+
+		klog.Infof("List of ApiResources fetched from apiserver: %s", string(printApiresourcesList))
+	}
+
+	return apiresourcesList, nil
+}
+
+func fetchDataActionsList(environment string, clusterType string, loginURL string, tenantID string, clientID string, clientSecret string) ([]Operation, error) {
```

Comment:
Instead of multiple utilities, should we explore moving this functionality to class?

Reply:
done moved the variables to a struct